api/src/routes/settings.test.ts

```diff
@@ -186,6 +186,158 @@ describe('settingRoutes', () => {
+          type: 'info'
+        });
+      });
+
+      test('PUT returns an error when the username is a https status code', async () => {
+        const response = await superRequest('/update-my-username', {
+          method: 'PUT',
+          setCookies
+        }).send({
+          username: '404'
+        });
+
+        expect(response?.statusCode).toEqual(400);
+
+        expect(response?.body).toEqual({
+          message: 'Username 404 is a reserved error code',
+          type: 'info'
+        });
+      });
+
+      test('PUT returns an error when the username is shorter than 3 characters', async () => {
+        const response = await superRequest('/update-my-username', {
+          method: 'PUT',
+          setCookies
+        }).send({
+          username: 'fo'
+        });
+
+        expect(response?.statusCode).toEqual(500);
+      });
+
+      test('PUT returns 200 status code with "success" message', async () => {
+        const response = await superRequest('/update-my-username', {
+          method: 'PUT',
+          setCookies
+        }).send({
+          username: 'twaha1'
+        });
+
+        expect(response?.statusCode).toEqual(200);
+
+        expect(response?.body).toEqual({
+          message: 'flash.username-updated',
+          type: 'success',
+          username: 'twaha1'
+        });
+      });
+
+      test('PUT returns an error when the username is already used', async () => {
+        await superRequest('/update-my-username', {
+          method: 'PUT',
+          setCookies
+        }).send({
+          username: 'twaha2'
+        });
+
+        const response = await superRequest('/update-my-username', {
+          method: 'PUT',
+          setCookies
+        }).send({
+          username: 'twaha2'
+        });
+
+        expect(response?.statusCode).toEqual(400);
+
+        expect(response?.body).toEqual({
+          message: 'flash.username-used',
+          type: 'info'
+        });
+      });
+
+      test('PUT returns 200 status code with "success" message', async () => {
+        await superRequest('/update-my-username', {
+          method: 'PUT',
+          setCookies
+        }).send({
+          username: 'twaha3'
+        });
+
+        const response = await superRequest('/update-my-username', {
+          method: 'PUT',
+          setCookies
+        }).send({
+          username: 'TWaha3'
+        });
+
+        expect(response?.statusCode).toEqual(200);
+
+        expect(response?.body).toEqual({
+          message: 'flash.username-updated',
+          type: 'success',
+          username: 'TWaha3'
+        });
+      });
+      test('PUT /update-my-username returns 400 status code when username is too long', async () => {
+        const username = 'a'.repeat(1001);
+        const response = await superRequest('/update-my-username', {
+          method: 'PUT',
+          setCookies
+        }).send({
+          username
+        });
+
+        expect(response?.statusCode).toEqual(500);
```

Comment:
```suggestion
        expect(response?.statusCode).toEqual(400);
```
Same principle here.